enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,199 @@
+---
+title: configurable-dns-for-loadbalancerservice
+authors:
+  - "@thejasn"
+reviewers:
+  - "@alebedev87"
+  - "@sherine-k"
+approvers:
+  - TBD
+api-approvers:
+  - TBD
+creation-date: 2022-05-10
+last-updated: 2022-05-10
+tracking-link:
+  - https://issues.redhat.com/browse/CFEPLAN-58
+see-also:
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# Configurable DNS Management for LoadBalancerService Ingress Controllers
+
+## Summary
+
+The ingress operator currently assumes that it's going to create and
+manage DNS using the required cloud provider integration, but in some
+circumstances when the DNS zone may be different from the cluster DNS zone
+and may not even hosted in the cloud provider. In such scenarios the operator
+after provisioning the required resources(load balancer, haproxy) in the cloud
+provider reports a degraded state because the provided DNS can't be created.
+
+This enhancement aims to provide the end-user the ability to completely disable
+DNS management on an Ingress Controller.
+
+## Motivation
+
+End users need to have an option to manage DNS on their own instead of relying
+on the `cluster-ingress-operator`. This is quite a common requirement where
+1. Customers may need deploy ingress controllers on domains hosted outside of
+   cloud providers.
+2. Partners may need to deploy ingress controllers for domains from their own 
+   customers (with domains not hosted in the same cloud provider).
+
+With no support for such scenarious currently, customers end up with clusters
+in degraded state. The solution to this is to add support
```

Comment:
I think that `degraded state` is just one of the possible results: when the DNS zone was not created by the installer.
The scenarios described above imply the usage of an external (to the cloud provider) DNS which doesn't exclude the existence of the cluster's DNS zone. So the controller may easily be in non degraded state but not willing to setup the DNS using the cluster's zone.

Reply:
I checked this out as well, looks like there is another issue where updating the [domain](https://github.com/openshift/cluster-ingress-operator/blob/972f09b9dbb181ae5c414da2a990b57c60fde9d8/pkg/operator/controller/ingress/controller.go#L342-L355) in the ingresscontroller doesn't result in [updating](https://github.com/openshift/cluster-ingress-operator/blob/972f09b9dbb181ae5c414da2a990b57c60fde9d8/pkg/operator/controller/ingress/dns.go#L97) the wildcard DNS.  

If we fix this issue, the dns_controller will end up with an error when it tries to publish the new record to the old zone. This would lead to the `DNSReadyIngressConditionType` to be set to false, resulting in degraded state.

Reply:
Yes, the degraded state is indeed quite a probable outcome if the above mentioned scenarios are tried to be implemented without the proposed solution. I think what poked into my eye is the wording `degraded cluster` while I thought more of `degraded ingress operator`.